src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/DbContextEditorServices.cs

```diff
@@ -171,6 +187,76 @@ public EditSyntaxTreeResult EditStartupForNewContext(ModelType startUp, string d
             };
         }
 
+        private IPropertySymbol TryGetIConfigurationRootProperty(ITypeSymbol startup)
+        {
+            var propertySymbols = startup.GetMembers()
+                .Select(m => m as IPropertySymbol)
+                .Where(s => s != null);
+
+            foreach (var pSymbol in propertySymbols)
+            {
+                var namedType = pSymbol.Type as INamedTypeSymbol; //When can this go wrong?
+                if (namedType != null &&
+                    namedType.ContainingAssembly.Name == "Microsoft.Extensions.Configuration.Abstractions" &&
+                    namedType.ContainingNamespace.ToDisplayString() == "Microsoft.Extensions.Configuration" &&
+                    namedType.Name == "IConfigurationRoot") // What happens if the type is referenced in full in code??
```

Comment:
Yes, I did test this manually and it does work irrespective if the property is defined by using IConfigurationRoot or Microsoft.Extensions.Configuration.IConfigurationRoot

---
In reply to: [66546977](https://github.com/aspnet/Scaffolding/pull/236#discussion_r66546977) [](ancestors = 66546977)